### PR TITLE
edit-util: a few cleanups; support networkctl edit --stdin

### PR DESCRIPTION
**Pull request by YHNdnzj**

### CODE REVIEW

File: src/shared/edit-util.c [hunk 1]
```diff
@@ -111,6 +111,7 @@ int edit_files_add(
 
 static int populate_edit_temp_file(EditFile *e, FILE *f, const char *filename) {
         assert(e);
```
Comment (bluca) [inline src/shared/edit-util.c:113]: assert(e->context)

File: man/networkctl.xml [hunk 2]
```diff
@@ -608,6 +611,23 @@ s - Service VLAN, m - Two-port MAC Relay (TPMR)
         </listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--stdin</option></term>
+
+        <listitem>
+          <para>When used with <command>edit</command>, the contents of the file will be read from standard
+          input and the editor will not be launched. In this mode, the old contents of the file are
+          automatically replaced. This is useful to "edit" configuration from scripts, especially that
```
Comment (bluca) [inline man/networkctl.xml]: "especially so that"

File: man/networkctl.xml [hunk 3]
```diff
@@ -608,6 +611,23 @@ s - Service VLAN, m - Two-port MAC Relay (TPMR)
...
+          drop-in directories are created and populated in one go.</para>
+
+          <para>Multiple drop-ins may be "edited" in this mode with <option>--drop-in=</option>, and
+          the same contents will be written to all of them. Otherwise exact one main configuration file
```
Comment (bluca) [inline man/networkctl.xml]: "exactly one"

File: man/networkctl.xml [hunk 4]
```diff
@@ -448,6 +448,9 @@ s - Service VLAN, m - Two-port MAC Relay (TPMR)
         <citerefentry><refentrytitle>systemd.link</refentrytitle><manvolnum>5</manvolnum></citerefentry>
         for more information.</para>
 
+        <para>If <option>--stdin</option> is specified, the new contents will be read from standard input.
```
Comment (bluca) [inline man/networkctl.xml]: "new content"

File: man/networkctl.xml [hunk 5]
```diff
@@ -448,6 +448,9 @@ s - Service VLAN, m - Two-port MAC Relay (TPMR)
...
+        In this mode, the old contents of the file are discarded.</para>
```
Comment (bluca) [inline man/networkctl.xml]: "old content of the file is discarded"

File: src/network/networkctl-config-file.c [hunk 6]
```diff
@@ -396,23 +396,30 @@ static int reload_daemons(ReloadFlags flags) {
...
         _cleanup_(sd_netlink_unrefp) sd_netlink *rtnl = NULL;
         ReloadFlags reload = 0;
         int r;
 
-        if (!on_tty())
-                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Cannot edit network config files if not on a tty.");
+        if (!on_tty() && !arg_stdin)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Cannot edit network config files interactively if not on a tty.");
+
+        /* Duplicate main configs make no sense. This also mimics the behavior of systemctl. */
```
Comment (bluca) [inline src/network/networkctl-config-file.c]: "Duplicating main configs makes"

**Review by bluca — APPROVED** [review]

**Comment by YHNdnzj** [comment]
Added a few more changes to the first commit. Please take another look.

**Review by yuwata — COMMENTED** [review]
LGTM.

**Pull request merged**